src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/ExcelExtractor.php

```diff
@@ -61,10 +61,12 @@ public function extract(FlowContext $context) : \Generator
 
         foreach ($context->streams()->list($this->path, $this->filter()) as $stream) {
             foreach ($this->extractRows($stream, $headers, $offset) as $row) {
-                $signal = yield array_to_rows($row, $context->entryFactory());
+                $signal = yield rows(array_to_row($row, $context->entryFactory(), $stream->path()->partitions()));
                 $this->incrementReturnedRows();
 
                 if ($signal === Signal::STOP || $this->reachedLimit()) {
+                    $this->reader()->close();
```

Comment:
just out of curiosity, what happens when you don't close the reader?

Reply:
btw I just noticed that you also need to remove `Closure` from Extractor, Extractors don't need to implement Closures.

Reply:
It leaves the stream open to block access to the file, and if you do a few DF in a row, it may eat memory.